src/icon.c

```diff
@@ -141,8 +172,20 @@ GdkPixbuf *get_pixbuf_from_file(const char *filename)
 {
         char *path = string_to_path(g_strdup(filename));
         GError *error = NULL;
+        gint w, h;
 
-        GdkPixbuf *pixbuf = gdk_pixbuf_new_from_file(path, &error);
+        if (!gdk_pixbuf_get_file_info (path, &w, &h)) {
+                LOG_W("Failed to load image info for %s", filename);
+                g_free(path);
+                g_error_free(error);
```

Comment:
Whoop, caught a bug! `error == NULL` always here, there is no need to free. (thankfully glib catches it and doesn't crash)

Reply:
:sweat_smile: